homeassistant/components/recorder/purge.py

```diff
@@ -195,3 +195,22 @@ def _purge_filtered_events(session: Session, excluded_event_types: list[str]) ->
     state_ids: list[int] = [state.state_id for state in states]
     _purge_state_ids(session, state_ids)
     _purge_event_ids(session, event_ids)
+
+
+@retryable_database_job("purge")
+def purge_entity_data(instance: Recorder, entity_filter: Callable[[str], bool]) -> bool:
+    """Purge states and events of specified entities."""
+    with session_scope(session=instance.get_session()) as session:  # type: ignore
+        # Purge a max of MAX_ROWS_TO_PURGE, based on the oldest states or events record
+        selected_entity_ids: list[str] = [
+            entity_id
+            for (entity_id,) in session.query(distinct(States.entity_id)).all()
+            if entity_filter(entity_id)
+        ]
+        _LOGGER.debug("Remove entity data for %s", selected_entity_ids)
+        if len(selected_entity_ids) > 0:
+            _purge_filtered_states(session, selected_entity_ids)
```

Comment:
Outside the scope of this PR, but is the function signature reversed on `_purge_filtered_states`

Should it be 

`def _purge_filtered_states(session: Session, included_entity_ids: list[str]) -> None:`?

Reply:
I think the original use-case for this was to "re-apply" the exclude list to the database i.e. as a retrospective, corrective action. Your change may be more intuitive for broader use.